services/shhext/api.go

```diff
@@ -441,3 +445,30 @@ func makePayload(r MessagesRequest) []byte {
 
 	return append(data, cursorBytes...)
 }
+
+func createBloomFilter(r MessagesRequest) []byte {
+	if len(r.Topics) > 0 {
+		return topicsToBloom(r.Topics...)
+	}
+
+	return whisper.TopicToBloom(r.Topic)
+}
+
+func topicsToBloom(topics ...whisper.TopicType) []byte {
```

Comment:
maybe OR the topic at every step and then copy after ORing all topics? smth like
```
rst = new(big.Int)
for _, t = range topic {
  rst.Or(rst, big.Int.SetBytes(topicToBloom))
}
bloom := make([]byte, 64)
copy()
return bloom
```

Reply:
@dshulyak yes that needed a refactoring from the initial version, thank you! I updated it with something like your snippet!